### PR TITLE
Append requestIdentifier in context data for Assurance and add tests

### PR DESCRIPTION
**Pull request by praveek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPAnalytics/Tests/AnalyticsTests.swift [hunk 1]
```diff
@@ -13,22 +13,231 @@
...
+            ([AnalyticsConstants.Configuration.GLOBAL_CONFIG_PRIVACY : PrivacyStatus.optedIn.rawValue], .set)
+        )
+        mockRuntime.simulateComingEvent(event: trackStateEvent)
+
+        XCTAssertEqual(mockRuntime.dispatchedEvents.count, 1)
+        let edgeEvent = mockRuntime.dispatchedEvents[0]
+        XCTAssertEqual(edgeEvent.type, EventType.edge)
+        XCTAssertEqual(edgeEvent.source, EventSource.requestContent)
+
+        let expectedData: [String:Any] = [
```
Comment (rymorale) [inline AEPAnalytics/Tests/AnalyticsTests.swift:49]: would it be worth it to create a function that takes as an input a bool signaling a trackState, trackAction, or trackInternalAction and a string containing the tracked action or state? the function could then return the expectedData for each test.

File: AEPAnalytics/Tests/TestHelper/TestUtils.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2020 Adobe. All rights reserved.
```
Comment (rymorale) [inline AEPAnalytics/Tests/TestHelper/TestUtils.swift:2]: change to 2021

File: AEPAnalytics/Tests/AnalyticsTests.swift (hunk 1 above)
Comment (praveek) [reply AEPAnalytics/Tests/AnalyticsTests.swift:49]: We just have one test for each API. I feel we can abstract the logic if we add more tests.

File: AEPAnalytics/Tests/TestHelper/TestUtils.swift (hunk 2 above)
Comment (praveek) [reply AEPAnalytics/Tests/TestHelper/TestUtils.swift:2]: I am reusing this from the edge extension.

**Review by rymorale — APPROVED** [review]
looks good, just had two comments that can be resolved another time.

File: AEPAnalytics/Tests/AnalyticsTests.swift (hunk 1 above)
Comment (rymorale) [reply AEPAnalytics/Tests/AnalyticsTests.swift:49]: sure sounds good.

**Pull request merged**